src/main/com/fulcrologic/guardrails/core.cljc

```diff
@@ -88,8 +88,8 @@
    (do
      (s/def ::defn-macro (s/nilable string?))
      (s/def ::expound (s/nilable (s/map-of keyword? any?)))
-     (s/def ::throw? (s/nilable (s/map-of keyword? any?)))
-     (s/def ::emit-spec? (s/nilable (s/map-of keyword? any?)))
+     (s/def ::throw? (s/nilable boolean?))
+     (s/def ::emit-spec? (s/nilable boolean?))
```

Comment:
I would also recommend removing `s/nilable` from all of these, the `opt-un` should be enough, in the `::guardrails-config`.

Also I see little reason to not check the config at compile time before caching for catching misspeccing like this in the future when the `clojure.spec.skip-macros` Java system property is not turned on.

Reply:
agreed.